### PR TITLE
ARTEMIS-2564 retryMessages incorrectly removes msgs

### PR DESCRIPTION
**Pull request by jbertram**

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java [hunk 1]
```diff
@@ -1996,8 +1980,7 @@ private int iterQueue(final int flushLimit,
                   }
 
                   if (filter1 == null || filter1.match(ref.getMessage())) {
-                     messageAction.actMessage(tx, ref);
-                     if (remove) {
+                     if (messageAction.actMessage(tx, ref)) {
```
Comment (brusdev) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java]: Why pageIterator loop [1] hasn't the same check?

[1] https://github.com/jbertram/activemq-artemis/blob/ARTEMIS-2564/artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java#L2015

Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java]: There's no need to do anything different.

We could have the same test performed over paging though.

Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java]: let me handle this in this afternoon. I already changed the test to use paging... but I can't make that part of the code to be used. 

leave it with me for a few hours.

Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java]: I give up... and there's no need for that..


I will merge it.. if anyone has a strong opinion of adding the test.. he's welcome to do it :)

Comment (brusdev) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java]: I created the PR #2914 to fix the removing of paged messages.

**Pull request closed without merging**